## Hypothesis testing

## Q1.Suppose a child psychologist claims that the average time working mothers spend talking to their children is at least 11 minutes per day. You conduct a random sample of 1000 working mothers and find they spend an average of 11.5 minutes per day talking with their children. Assume prior research suggests the population standard deviation is 2.3 minutes.Conduct a test with a level of significance of alpha = 0.05.

In [1]:
pip install statsmodels

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 23.2.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip


In [2]:
pip install scipy

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 23.2.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip


In [3]:
import numpy as np

## NULL HYPOTHESIS

Ho = μ ≤ 11 minutes

## ALTERNATIVE HYPOTHESIS

Ha = μ > 11 minutes

x̄ = 11.5 minute

s
n = 10

00
σ = 2.3 minu

tes
α = 0.05

## Statistical Test

## 1.Test Selection:

Use a Z-test because the sample size (1000) is greater than 3
le size): 1000

## 2.Rejection Criteria:

Reject ( H_0 ) if ( z_{statistics} > ) critical value.


## 3.Z-Test Formula:

( Z = \frac{\bar{x} - \mu}{\sigma / \sqrt{n}} )


## 4.Parameters:

( \bar{x} ) (sample mean): 11.5 minutes

( \mu ) (population mean): 11 minutes

( \sigma ) (population standard deviation): 2.3 minutes

( n ) (sample size): 1000

In [5]:
import math

In [6]:
Z = (11.5-11)/(2.3/math.sqrt(1000))
Z

6.874516652539955

## Critical Value/P-Value:

Since our calculated z-score (6.87) is much greater than the critical value (1.645), we reject the null hypothesis. We have sufficient evidence to conclude that the average time working mothers spend talking to their children is more than 11 minutes per day .

## Q2. A coffee shop claims that their average wait time for customers is less than 5 minutes. To test this claim, a sample of 40 customers is taken, and their wait times are recorded. The sample mean wait time is found to be 4.6 minutes with a standard deviation of 0.8 minutes. Perform a hypothesis test at a significance level of 0.05 and determine whether there is enough evidence to support the coffee shop's claim.

Ho = μ ≥ 5 minute (Null Hypothesis)sv in-1) = 39 Ha = μ < 5 minutes (Alternative Hypothesis)


x̄ = 4.6 minutes

n = 40

σ = 0.8 minutes

μ = 5 minutes

α = 0.05(5%)

n-1 = 39

## Statistical Test:

If the sample size is small (typically <30) and the population standard deviation is unknown, the t-test is appropriate. However, for larger sample sizes (usually greater than 30), the t-test can also be used even if the population standard deviation is unknown due to the Central Limit Theorem, which states that the distribution of sample means approaches a normal distribution as the sample size increases.

t_test_statistic=(x̄ - μ) / (σ / √n)

x̄ = sample mean (4.6 minutes)

μ = population mean (5 minutes)

σ = Sample standard deviation (0.8 minutes)

n = sample size (40).

α = Significance level (0.05(5%)).

n-1 = Degree of freedom(39).

In [7]:
t_test_statistic=(4.6 - 5) / (0.8 / np.sqrt(40))
print('t_statistic:',t_test_statistic)

t_statistic: -3.162277660168382


In [8]:
#calculating the critical value with df 39 and alpha 0.05
from scipy.stats import t
critical_value = t.ppf(0.05, 39)
print("Critical value:", critical_value)

Critical value: -1.684875121711225


## Conclusion:
Since our calculated t_test(-3.16) is less than the critical t-value (-1.685), we fail to accept the null hypothesis. There is enough evidence to support the claim that the average wait time for customers is less than 5 minutes at a significance level of 0.05.